### DISCUSSION TITLE
Problems while extracting a portion of an image using a mask.

### DISCUSSION THREAD

**Discussion by nriesco** [ImageMagick/ImageMagick#4101] [Help]
Hi, 

I successfully extracted a certain portion of a big image using a mask, with this code:
```
convert bigImage.jpeg mask.png \
  \( -clone 0 -alpha extract \) \
  \( -clone 1,2 -compose multiply -composite \) \
  -delete 1,2 -compose over -compose copy_opacity -composite result.png
```

This results of the mask being applied on the top-left corner.

The problem is that the mask could be somewhere else (relative to the bigImage) and could algo be smaller. So I need to modify this command in order to "move" the mask before applying it.

Any idea on how to do this?
Thanks!

**Important note:** I don't fully understand the code I'm using, I started from an example on the web

**Comment by fmw42** [comment]
Please post an image example showing what you have and what you want to change.

**Reply by fmw42** [reply]
Your code is taking the alpha channel from the image, combining it with the mask to expand the area of transparency and then putting the combined alpha channel back into the image.

**Reply by nriesco** [reply]
These are the files I'm using.

So what I would like to achieve?
Running the same command with some modification so it won't get the top-left corner of bigImage but say the center, or x pixels to the right.

The desired output is shown (approximately) in `desiredOutputAprox.png`

![bigImage](https://user-images.githubusercontent.com/1995514/130334545-c344b335-9d5b-4d1b-befe-c49203934c89.jpeg)

![mask](https://user-images.githubusercontent.com/1995514/130334547-8de15622-6a00-4a6b-bcc5-2c9ac49f6539.png)

![result](https://user-images.githubusercontent.com/1995514/130334548-0351ce87-51d5-4853-a4fa-5f070d4c83b2.png)

![desiredOutputAprox](https://user-images.githubusercontent.com/1995514/130334551-fc408bb9-e910-4f65-9c89-b458c742e629.png)

[maskExample.zip](https://github.com/ImageMagick/ImageMagick/files/7026199/maskExample.zip)

**Comment by fmw42** [comment]
Here is one way using Imagemagick 7 on Unix-like systems.

Define the point to be the center of region you want. Extract the alpha channel of the mask and get its dimensions. Then crop the large image at that size and offset from its center. Then put the mask alpha channel into the alpha channel of the cropped image.

```
xloc=740
yloc=330
im7 magick \( mask.png +repage -alpha extract -set option:dims "%wx%h" \) \
\( bigImage.jpeg -gravity center  \
-crop "%[dims]+%[fx:$xloc-w/2]+%[fx:$yloc-h/2]" +repage \) \
+swap -compose copy_opacity -composite \
result.png

```

![result](https://user-images.githubusercontent.com/13458218/130335232-ad926894-538b-42eb-a64f-b4a4d3217edf.png)

**Reply by nriesco** [reply]
Wow!!! thanks again @fmw42 !!

Actually this not only does what I asked, it does the whole thing plus the next step (clipping) in a fraction of the time. I really appreciate it, thanks!!!

BTW: I still cannot mark as answered any of these questions due to a but in github's discussion (see https://github.com/github/feedback/discussions/5228)

**Comment by snibgo** [comment]
fmw42 wrote:
> -crop "%[dims]+%[fx:$xloc-w/2]+%[fx:$yloc-h/2]+%[fx:$yloc-h/2]" +repage \) \

There seems to be a typo, with an extra "+%[fx:$yloc-h/2]".

**Comment by fmw42** [comment]
Not sure how that happened. Fixed it. Thanks.
